src/java/org/apache/cassandra/service/accord/FetchTopology.java

```diff
@@ -123,18 +116,20 @@ public Response(long epoch, Topology topology)
         long epoch = message.payload.epoch;
         Topology topology = AccordService.instance().topology().maybeGlobalForEpoch(epoch);
         if (topology == null)
-            MessagingService.instance().respond(Response.UNKNOWN, message);
+            MessagingService.instance().respond(Response.unkonwn(epoch), message);
```

Comment:
im confused by this line

```
        private static Response unkonwn(long epoch)
        {
            throw new IllegalStateException("Unknown topology: " + epoch);
        }
```

`Response.unkonwn` only throws, so this `response` won't do anything as that method won't be called

Reply:
Is this for test code that can catch the actual exception?

Reply:
i couldn't find any tests for this logic.  Since this is a verb handle we should get back an `UNKNOWN` exception; that could be a NPE or an unknown epoch... its truly unknown.

Reply:
switched to an explicit throw so that we would return UNKNOWN via messaging.

Reply:
I don't feel this is a good idea.... `UNKNOWN` can also mean NPE... its truly unknown whats going on and isn't a good answer to "do you know about this epoch".  For example, if the epoch isn't known there isn't a reason to retry on that node... but with `UNKNOWN` error we then will retry

Reply:
created https://github.com/apache/cassandra/pull/3842#discussion_r1936394993 so this doesn't get lost...